decorators/nameable.rb

```diff
@@ -0,0 +1,31 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented the method '#{__method__}'"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    curr_name = @nameable.correct_name
+    return curr_name if curr_name.length <= 10
+
+    curr_name.slice(0, 10)
```

Comment:
- [ ]  **Hello, I would advise that each class go in its own file. it is easier to navigate large classes when they are separated**. Here is the [the big picture of this project](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/sneak_peek.md).  Look at the kind of application you are developing, your code needs to be structured so that it is easy to understand, document, and modify.